src/main/java/at/grahsl/kafka/connect/mongodb/MongoDbSinkConnectorConfig.java

```diff
@@ -120,7 +123,52 @@ public MongoDbSinkConnectorConfig(Map<String, String> parsedConfig) {
     }
 
     public static ConfigDef conf() {
-        return new ConfigDef()
+        return new ConfigDef() {
+
+            <T> Validation<T> ensureValid(String name, Consumer<T> consumer) {
+                return new Validation<T>(name, consumer);
+            }
+
+            class Validation<T> {
+
+                private final String name;
+                private final Consumer<T> consumer;
+
+                Validation(String name, Consumer<T> consumer) {
+                    this.name = name;
+                    this.consumer = consumer;
+                }
+
+                Validation<T> unless(boolean condition) {
+                    return condition ? new Validation<T>(name, (T t) -> {}) : this;
+                }
+            }
+
+            @Override
+            public Map<String, ConfigValue> validateAll(Map<String, String> props) {
+                Map<String, ConfigValue> result = super.validateAll(props);
+                MongoDbSinkConnectorConfig config = new MongoDbSinkConnectorConfig(props);
```

Comment:
If we validate 'most critical' config settings already it might make sense to validate all available ones? what do you think?

Reply:
I took another close look at what is not validated in my proposal.

Here is a list of parameters NOT directly validated. They break down into three groups:

Primitive types validated in superclass:
* MONGODB_MAX_NUM_RETRIES_CONF :: validated as int type in superclass
* MONGODB_RETRIES_DEFER_TIMEOUT_CONF :: validated as int type in superclass
* MONGODB_DELETE_ON_NULL_VALUES :: validated as boolean in superclass
These are fine.

Validation of list elements:
* MONGODB_VALUE_PROJECTION_LIST_CONF :: validated as part of MONGODB_VALUE_PROJECTION_TYPE_CONF
* MONGODB_DOCUMENT_ID_STRATEGY_CONF :: validated as part of MONGODB_DOCUMENT_ID_STRATEGIES_CONF validation
* MONGODB_KEY_PROJECTION_LIST_CONF :: validated as part of MONGODB_KEY_PROJECTION_TYPE_CONF

Kind of dirty... It might be a good idea to create a syntactical validator to be used inside `.define`, i.e. a syntactical validator for classnames and comma-separated lists thereof, in addition to the implicit semantical validation in `.ensureValid`. Apart from that, I don't think it is worth the effort to validate all paths of execution. 

Not validated at all:
* MONGODB_COLLECTION_CONF

Should the connection be tested during validation already? I am not sure about that.

Reply:
I think we can for now safely skip to validate the connection. But to have a better way to semantically validate different list element type config settings would definitely nice in future versions. feel free to keep me updated in case you want to get things done in this regard!